File: Cryambly/Engine/Logic/Entities/EntityAspects.cs
namespace CryCil.Engine.Logic
{
	/// <summary>
	/// Enumeration of aspects of the entity that can be synchronized via network.
	/// </summary>
	public enum EntityAspects : uint
	{
		/// <summary>
		/// Represents all aspects. Used to ensure all aspects are synchronized at once.
		/// </summary>
		All = uint.MaxValue,
		// 0x01u // aspect 0

		/// <summary>
		/// Script aspect. Not used in CryCIL.
		/// </summary>
		Script = 0x02u,
		// 0x04u // aspect 2

		/// <summary>
		/// Physics aspect.
		/// </summary>
		Physics = 0x08u,
		/// <summary>
		/// </summary>
		GameClientStatic = 0x10u, // aspect 4
		/// <summary>
		/// </summary>
		GameServerStatic = 0x20u, // aspect 5
		/// <summary>
		/// </summary>
		GameClientDynamic = 0x40u, // aspect 6
		/// <summary>
		/// </summary>
		GameServerDynamic = 0x80u, // aspect 7
		/// <summary>
		/// </summary>
		GameClientA = 0x0100u, // aspect 8
		/// <summary>
		/// </summary>
		GameServerA = 0x0200u, // aspect 9
		/// <summary>
		/// </summary>
		GameClientB = 0x0400u, // aspect 10
		/// <summary>
		/// </summary>
		GameServerB = 0x0800u, // aspect 11
		/// <summary>
		/// </summary>
		GameClientC = 0x1000u, // aspect 12
		/// <summary>
		/// </summary>
		GameServerC = 0x2000u, // aspect 13
		/// <summary>
		/// </summary>
		GameClientD = 0x4000u, // aspect 14
		/// <summary>
		/// </summary>
		GameClientE = 0x8000u, // aspect 15
		/// <summary>
		/// </summary>
		GameClientF = 0x00010000u, // aspect 16
		/// <summary>
		/// </summary>
		GameClientG = 0x00020000u, // aspect 17
		/// <summary>
		/// </summary>
		GameClientH = 0x00040000u, // aspect 18
		/// <summary>
		/// </summary>
		GameClientI = 0x00080000u, // aspect 19
		/// <summary>
		/// </summary>
		GameClientJ = 0x00100000u, // aspect 20
		/// <summary>
		/// </summary>
		GameServerD = 0x00200000u, // aspect 21
		/// <summary>
		/// </summary>
		GameClientK = 0x00400000u, // aspect 22
		/// <summary>
		/// </summary>
		GameClientL = 0x00800000u, // aspect 23
		/// <summary>
		/// </summary>
		GameClientM = 0x01000000u, // aspect 24
		/// <summary>
		/// </summary>
		GameClientN = 0x02000000u, // aspect 25
		/// <summary>
		/// </summary>
		GameClientO = 0x04000000u, // aspect 26
		/// <summary>
		/// </summary>
		GameClientP = 0x08000000u, // aspect 27
		/// <summary>
		/// </summary>
		GameServerE = 0x10000000u, // aspect 28
		/// <summary>
		/// </summary>
		Aspect29 = 0x20000000u, // aspect 29
		/// <summary>
		/// </summary>
		Aspect30 = 0x40000000u, // aspect 30
		/// <summary>
		/// </summary>
		Aspect31 = 0x80000000u // aspect 31
	}
}

File: MonoInterface/TypeMapChecks/EntityAspects.cpp
#include "stdafx.h"

#include "CheckingBasics.h"

#include "IGameObject.h"

TYPE_MIRROR enum EntityAspects
{
	eEA_All_check = NET_ASPECT_ALL,
	// 0x01u                       // aspect 0
	eEA_Script_check = 0x02u, // aspect 1
	// 0x04u                       // aspect 2
	eEA_Physics_check = 0x08u, // aspect 3
	eEA_GameClientStatic_check = 0x10u, // aspect 4
	eEA_GameServerStatic_check = 0x20u, // aspect 5
	eEA_GameClientDynamic_check = 0x40u, // aspect 6
	eEA_GameServerDynamic_check = 0x80u, // aspect 7
	eEA_GameClientA_check = 0x0100u, // aspect 8
	eEA_GameServerA_check = 0x0200u, // aspect 9
	eEA_GameClientB_check = 0x0400u, // aspect 10
	eEA_GameServerB_check = 0x0800u, // aspect 11
	eEA_GameClientC_check = 0x1000u, // aspect 12
	eEA_GameServerC_check = 0x2000u, // aspect 13
	eEA_GameClientD_check = 0x4000u, // aspect 14
	eEA_GameClientE_check = 0x8000u, // aspect 15
	eEA_GameClientF_check = 0x00010000u, // aspect 16
	eEA_GameClientG_check = 0x00020000u, // aspect 17
	eEA_GameClientH_check = 0x00040000u, // aspect 18
	eEA_GameClientI_check = 0x00080000u, // aspect 19
	eEA_GameClientJ_check = 0x00100000u, // aspect 20
	eEA_GameServerD_check = 0x00200000u, // aspect 21
	eEA_GameClientK_check = 0x00400000u, // aspect 22
	eEA_GameClientL_check = 0x00800000u, // aspect 23
	eEA_GameClientM_check = 0x01000000u, // aspect 24
	eEA_GameClientN_check = 0x02000000u, // aspect 25
	eEA_GameClientO_check = 0x04000000u, // aspect 26
	eEA_GameClientP_check = 0x08000000u, // aspect 27
	eEA_GameServerE_check = 0x10000000u, // aspect 28
	eEA_Aspect29_check = 0x20000000u, // aspect 29
	eEA_Aspect30_check = 0x40000000u, // aspect 30
	eEA_Aspect31_check = 0x80000000u, // aspect 31
};

#define CHECK_ENUM(x) static_assert (EntityAspects::x ## _check == EEntityAspects::x, "EEntityAspects enumeration has been changed.")

inline void Check()
{
	CHECK_ENUM(eEA_All);
	CHECK_ENUM(eEA_Script);
	CHECK_ENUM(eEA_Physics);
	CHECK_ENUM(eEA_GameClientStatic);
	CHECK_ENUM(eEA_GameServerStatic);
	CHECK_ENUM(eEA_GameClientDynamic);
	CHECK_ENUM(eEA_GameServerDynamic);
	CHECK_ENUM(eEA_GameClientA);
	CHECK_ENUM(eEA_GameServerA);
	CHECK_ENUM(eEA_GameClientB);
	CHECK_ENUM(eEA_GameServerB);
	CHECK_ENUM(eEA_GameClientC);
	CHECK_ENUM(eEA_GameServerC);
	CHECK_ENUM(eEA_GameClientD);
	CHECK_ENUM(eEA_GameClientE);
	CHECK_ENUM(eEA_GameClientF);
	CHECK_ENUM(eEA_GameClientG);
	CHECK_ENUM(eEA_GameClientH);
	CHECK_ENUM(eEA_GameClientI);
	CHECK_ENUM(eEA_GameClientJ);
	CHECK_ENUM(eEA_GameServerD);
	CHECK_ENUM(eEA_GameClientK);
	CHECK_ENUM(eEA_GameClientL);
	CHECK_ENUM(eEA_GameClientM);
	CHECK_ENUM(eEA_GameClientN);
	CHECK_ENUM(eEA_GameClientO);
	CHECK_ENUM(eEA_GameClientP);
	CHECK_ENUM(eEA_GameServerE);
	CHECK_ENUM(eEA_Aspect29);
	CHECK_ENUM(eEA_Aspect30);
	CHECK_ENUM(eEA_Aspect31);
}